contracts/index_executor/src/task.rs

```diff
@@ -257,6 +321,90 @@ impl Task {
         Ok(())
     }
 
+    pub fn apply_recipient(&mut self, context: &Context) -> Result<(), &'static str> {
+        let step_count = self.steps.len();
+        for (index, step) in self.steps.iter_mut().enumerate() {
+            let step_source_chain = &step.source_chain(context).ok_or("MissingChain")?;
+            let step_dest_chain = &step.dest_chain(context).ok_or("MissingChain")?;
+            let worker_account_info = context.get_account(self.worker).ok_or("WorkerNotFound")?;
+
+            // For sure last step we should put real recipient, or else the recipient could be either
```

Comment:
According to our previous discussion, all the recipient should be the worker account, unless it uses the handler to temporarily control the assets for swap and bridge, right?

Reply:
yes, considering this need refactor on the smart contract side (support batch call along with asset transfer from worker to handler), I will make a separate PR